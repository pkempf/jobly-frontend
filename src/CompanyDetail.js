import React, { useContext, useState } from "react";
import { useParams } from "react-router-dom";
import Container from "react-bootstrap/Container";
import UserContext from "./UserContext";
import JoblyApi from "./api";

// import CompanyCard from "./CompanyCard";

const CompanyDetail = () => {
  const { handle } = useParams();
  const user = useContext(UserContext);

  return (
    <Container className="CompanyDetail">
      <p>The company detail will go here.</p>
    </Container>
  );
};

export default CompanyDetail;
