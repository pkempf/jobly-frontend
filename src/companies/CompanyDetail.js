import React, { useContext, useState, useEffect } from "react";
import { useParams, Redirect } from "react-router-dom";
import Container from "react-bootstrap/Container";
import UserContext from "../UserContext";
import JobCard from "../jobs/JobCard";
import JoblyApi from "../api";

// import CompanyCard from "./CompanyCard";

const CompanyDetail = () => {
  const { handle } = useParams();
  const user = useContext(UserContext);
  const [company, setCompany] = useState({});

  useEffect(() => {
    let isRendered = true;

    async function getCompany(companyHandle) {
      try {
        if (isRendered && user.username) {
          let companyRes = await JoblyApi.getCompany(companyHandle);
          let indicatedJobs = await JoblyApi.indicateAppliedJobs(
            companyRes.jobs,
            user.username
          );

          setCompany({ ...companyRes, jobs: indicatedJobs });
        }
      } catch (e) {
        console.log(e);
      }
    }
    getCompany(handle);

    return () => {
      isRendered = false;
    };
  }, [user.username, handle]);

  async function apply(job) {
    const applyRes = await JoblyApi.applyForJob(user.username, job.id);

    if (applyRes.applied && applyRes.applied === job.id) {
      let updatedJobs = [...company.jobs];
      let idx = company.jobs.findIndex((j) => j.id === job.id);

      let updatedJob = {
        ...job,
        applied: true,
      };

      updatedJobs[idx] = updatedJob;

      setCompany({ ...company, jobs: updatedJobs });
    }
  }

  if (!user.username) {
    return <Redirect to="/" />;
  }

  return (
    <Container className="CompanyDetail">
      <h2>{company.name}</h2>
      <p>{company.description}</p>
      {company.jobs && company.jobs.length > 0 ? (
        company.jobs.map((job) => (
          <JobCard job={job} key={job.id} apply={() => apply(job)} />
        ))
      ) : (
        <p>No matching jobs found.</p>
      )}
    </Container>
  );
};

export default CompanyDetail;
