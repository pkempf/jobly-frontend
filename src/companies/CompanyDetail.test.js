import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import CompanyDetail from "./CompanyDetail";
import UserContext from "../UserContext";

it("renders without crashing", () => {
  render(
    <MemoryRouter>
      <UserContext.Provider value={{ username: "test" }}>
        <CompanyDetail />
      </UserContext.Provider>
    </MemoryRouter>
  );
});

it("matches the snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <UserContext.Provider value={{ username: "test" }}>
        <CompanyDetail />
      </UserContext.Provider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
