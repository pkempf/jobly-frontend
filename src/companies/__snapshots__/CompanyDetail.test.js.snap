// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="CompanyDetail container"
  >
    <h2 />
    <p />
    <p>
      No matching jobs found.
    </p>
  </div>
</DocumentFragment>
`;
